### PR TITLE
8296198: JFileChooser throws InternalError java.lang.InternalError with Windows shortcuts

### PR DESCRIPTION
**Pull request by TejeshR13**
When `JFileChooser` uses custom `FileSystemView`, traversing to link/shortcut to a folder throw `InternalError java.lang.InternalError ` in Windows. The issue found out to be in BasicFileChooserUI class which was unable to resolve the link path during directory setting. The `UsesShellFolder` check expects the FileSystemView to be of WindowsFileSystemView, which fails when custom FileSystemView is used.
The fix includes resolving the link folder when custom File System View is used and has manual test. (Fix is test in CI system and no regression caused).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8296198](https://bugs.openjdk.org/browse/JDK-8296198): JFileChooser throws InternalError java.lang.InternalError with Windows shortcuts

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to [4dd8cc71](https://git.openjdk.org/jdk/pull/11510/files/4dd8cc71c550b56e7eec04cabcf5892aeabd23fd)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11510/head:pull/11510` \
`$ git checkout pull/11510`

Update a local copy of the PR: \
`$ git checkout pull/11510` \
`$ git pull https://git.openjdk.org/jdk pull/11510/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11510`

View PR using the GUI difftool: \
`$ git pr show -t 11510`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11510.diff">https://git.openjdk.org/jdk/pull/11510.diff</a>

</details>

### CODE REVIEW

File: src/java.desktop/share/classes/javax/swing/plaf/basic/BasicFileChooserUI.java [hunk 1]
```diff
@@ -1408,26 +1408,42 @@ public void actionPerformed(ActionEvent e) {
...
+                                return;
+                            }
+                        } else {
+                            dir = shellFolder;
+                        }
+                    }
+                } catch (FileNotFoundException ex) {
+                    return;
+                }
+            } else if ( fc.getFileSystemView().isLink(dir)) {
```
Comment (mrserb) [inline src/java.desktop/share/classes/javax/swing/plaf/basic/BasicFileChooserUI.java:1433]: Probably we can minimize the code duplication here?

File: test/jdk/javax/swing/JFileChooser/FileSystemView/CustomFSVLinkTest.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        }
+
+        @Override
+        public Icon getSystemIcon(File f) {
+            return delegate.getSystemIcon(f);
+        }
+        /*
+                @Override
+                public Icon getSystemIcon(File f, int width, int height) {
+                    return delegate.getSystemIcon(f, width, height);
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JFileChooser/FileSystemView/CustomFSVLinkTest.java:131]: Commented block of code can be removed.

Comment (TejeshR13) [reply test/jdk/javax/swing/JFileChooser/FileSystemView/CustomFSVLinkTest.java]: Updated.

File: src/java.desktop/share/classes/javax/swing/plaf/basic/BasicFileChooserUI.java (hunk 1 above)
Comment (TejeshR13) [reply src/java.desktop/share/classes/javax/swing/plaf/basic/BasicFileChooserUI.java]: Updated.

**Review by mrserb — APPROVED** [review]

File: test/jdk/javax/swing/JFileChooser/FileSystemView/CustomFSVLinkTest.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+
+import javax.swing.*;
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JFileChooser/FileSystemView/CustomFSVLinkTest.java:35]: The import's can be specific.

**Review by kumarabhi006 — APPROVED** [review]
Verified the test. Looks good to me.

**Review by kumarabhi006 — APPROVED** [review]

**Pull request closed without merging**